golem-ts-sdk/src/internal/clientGeneration.ts

```diff
@@ -108,18 +108,27 @@ function getMethodProxy(
         })()
       : parameterWitValuesEither.right;
 
-    const rpcForInvokeMethod = new WasmRpc(workerId);
+    const wasmRpc = new WasmRpc(workerId);
 
-    const rpcResult = rpcForInvokeMethod.invokeAndAwait(
+    const rpcResultFuture = wasmRpc.asyncInvokeAndAwait(
       functionName,
       parameterWitValues,
     );
 
+    const rpcResult = rpcResultFuture.get();
```

Comment:
As discussed in private, this needs an await `rpcResultFuture.promise()` too

Reply:
More precisely:

- call `rpcResultFuture.subscribe()` to get a `Pollable` value
- call `pollable.promise()` to get a JS promise
- await that
- then call `rpcResultFuture.get()` to get the value